plugins/mmdblookup/mmdblookup.c

```diff
@@ -133,6 +138,48 @@ void close_mmdb(MMDB_s *mmdb) {
 	MMDB_close(mmdb);
 }
 
+const char *SEP = "!";
+
+int mmdb_get_value(
+	MMDB_entry_s *const start,
+	MMDB_entry_data_s *const entry_data,
+	char *field,
+	size_t field_len
+) {
+	// Count names in the field to construct a path, e.g.,
+	// "!continent!code" implies 2 levels of JSON object {"continent": {"code": "XX"}}.
+	size_t length = 0;
+	for (size_t i = 0; i < field_len; i++) {
+		if (field[i] == *SEP) {
+			length++;
+		}
+	}
+
+	if (length == SIZE_MAX) {
```

Comment:
Why are we checking this? Won't it always be less than `field_len`? If we were in danger of hitting it, it seems like we would just overflow most of the time anyway.

Reply:
I based it on https://github.com/maxmind/libmaxminddb/blob/main/src/maxminddb.c#L1152. Basically this  `mmdb_get_value()` function is a variation of `MMDB_vget_value()` that can prepare the path based on the rsyslog's config field.